src/modules/importExport/utils/includeExclude.ts

```diff
@@ -0,0 +1,12 @@
+import { match, P } from "ts-pattern";
+
+export type IncludeExclude<Choices extends string> =
+  | { include: ReadonlyArray<Choices>; exclude?: undefined }
+  | { exclude: ReadonlyArray<Choices>; include?: undefined } // Exclude is optional because it's possible to use the type without
+  | {};
```

Comment:
Don't use this type (`{}`). Anything is assignable to it (e.g. try this `const shit: IncludeExclude<"a" | "b"> = 88;`, it doesn't show an error). You shouldn't need it here. Just make `include`/`exclude` optional in both cases. It will even work with the `ts-pattern` condition bellow.

Reply:
And make the type readonly, please. 😄

Reply:
You forgot a comment there that doesn't seem relevant anymore.